src/handlers/handleCreateRegOptions.ts

```diff
@@ -1,14 +1,66 @@
 import { Context } from 'hono';
+import { HTTPException } from 'hono/http-exception';
 import { zfd } from 'zod-form-data';
+import { generateRegistrationOptions } from '@simplewebauthn/server';
+import { cose, isoBase64URL } from '@simplewebauthn/server/helpers';
 
-import { regOptionsInputSchema } from '../schemas';
+import { ZodError, regOptionsInputSchema } from '../schemas';
 
 
 /**
  * Generate registration options
  */
 export async function handleCreateRegOptions(context: Context): Promise<Response> {
-	const parsedInput = zfd.formData(regOptionsInputSchema).parse(context.req.query());
-	console.log(parsedInput);
-	return context.text(`handleCreateRegOptions ${JSON.stringify(parsedInput)}`);
+	// Parse options from query params
+	let parsedInput;
+	try {
+		parsedInput = zfd.formData(regOptionsInputSchema).parse(context.req.query());
+	} catch (err) {
+		const _err = err as ZodError;
+		throw new HTTPException(400, { message: JSON.stringify(_err.errors) });
+	}
+
+	const {
+		algES256,
+		algRS256,
+		attestation,
+		discoverableCredential,
+		userVerification,
+		userName,
+		userID,
+		attachment,
+	} = parsedInput;
+
+	const { RP_ID, RP_NAME } = context.env;
+
+	let supportedAlgorithmIDs = [];
+
+	if (algES256) {
+		supportedAlgorithmIDs.push(cose.COSEALG.ES256);
+	}
+
+	if (algRS256) {
+		supportedAlgorithmIDs.push(cose.COSEALG.RS256);
+	}
+
+	let userIDBytes = undefined;
+	if (userID) {
+		userIDBytes = isoBase64URL.toBuffer(userID);
+	}
+
+	const opts = await generateRegistrationOptions({
+		rpID: RP_ID,
+		rpName: RP_NAME,
+		userName,
```

Comment:
```suggestion
		userName: username,
```

This threw me being case sensitive and wasn't expecting a capital N